test/e2e/handler/utils.go

```diff
@@ -123,6 +124,12 @@ func updateDesiredStateAtNodeAndWait(node string, desiredState nmstate.State) {
 	waitForAvailableTestPolicy()
 }
 
+func desiredStateWithDefaults(desiredState nmstate.State) nmstate.State {
```

Comment:
Please create a manual hard coded version of the desired state.
The e2e tests shouldn't use the same code the dev is using for verification. Bug in the dev code won't be discovered this way.